README.md

```diff
@@ -1,39 +1,124 @@
-# YOURPROJECTNAME
+# Hierarchy
 
-Describe what this project is all about.
+[![Build Status](https://travis-ci.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter.svg?branch=master)](https://travis-ci.com/KohlsTechnology/hierarcbhy)
+[![Go Report Card](https://goreportcard.com/badge/github.com/KohlsTechnology/prometheus_bigquery_remote_storage_adapter)](https://goreportcard.com/report/github.com/KohlsTechnology/hierarcbhy)
+
+Hierarchy is a simple utility to merge a set of yaml or json files, based on a defined hierarchy. It is inspired by Hiera from Puppet.
 
 ## Installation
 
-Add steps for installing here.
+You can either download a binary file from GitHub releases or [compile from source](#compiling-from-source).
 
 ## Documentation
 
-Add documentaiton here, or add a link to
-the full documentation.
+The main goal of `hierarchy` is to prevent the configuration data duplication and allow for a fine-grained control over the GitOps process. The output YAML can be used to generate a `values.yml` file for GitOps tools, like [Eunomia](https://github.com/KohlsTechnology/eunomia), [Helm](https://helm.sh), and others.
+
+### Usage
+
+You can control the behavior of Hierarchy either through command line options or environment variables. The later is especially helpful if you're running it inside a container.
+
+| Command Line Flag | Environment Variable | Default | Description |
+| --- | --- | --- | --- | --- |
+| `-f, --file` | `HIERARCHY_FILE` | `./hierarchy.lst` | Path and name of the hierarchy file. |
+| `-o, --output` | `HIERARCHY_OUTPUT` | `./output.yaml` | Path and name of the output file. |
+| `-i, --filter` | `HIERARCHY_FILTER` | `(.yaml|.yml|.json)$` | Regex for file extension of files being merged |
+| `-m, --failmissing` | `HIERARCHY_FAILMISSING` | `false` | Fail if a directory in the hierarchy is missing |
```

Comment:
```suggestion
| `-m, --failmissing` | `HIERARCHY_FAILMISSING` | `false` | Fail if a directory in the hierarchy is missing. |
```

Reply:
Also, I would think we should default this to true, but I may be missing something